pageserver/src/storage_sync2.rs

```diff
@@ -659,15 +668,61 @@ impl RemoteTimelineClient {
         Ok(())
     }
 
+    /// Schedule a batch of layer file uploads with subsequent IndexPart upload.
+    /// No-op if layer_files is empty.
+    /// If an error is returned, it is undefined which upload operations were scheduled
+    /// and wich ones weren't.
+    pub fn schedule_layer_file_uploads(
+        self: &Arc<Self>,
+        layer_files: &HashMap<LayerFileName, LayerFileMetadata>,
+    ) -> anyhow::Result<()> {
+        if layer_files.is_empty() {
+            return Ok(());
+        }
+
+        let mut guard = self.upload_queue.lock().unwrap();
+        let upload_queue = guard.initialized_mut()?;
+
+        for (name, md) in layer_files {
+            self.schedule_layer_file_upload_impl(upload_queue, name, md)
+                .with_context(|| {
+                    format!("scheduler layer file upload of file {}", name.file_name())
+                })?;
+        }
+
+        let metadata_bytes = upload_queue.latest_metadata.to_bytes()?;
+        let disk_consistent_lsn = upload_queue.latest_metadata.disk_consistent_lsn();
+        let index_part = IndexPart::new(
+            upload_queue.latest_files.clone(), // updated by schedule_layer_file_upload_impl
+            disk_consistent_lsn,
+            metadata_bytes,
+        );
+
+        let op = UploadOp::UploadMetadata(index_part, disk_consistent_lsn);
+        self.update_upload_queue_unfinished_metric(1, &op);
+        upload_queue.queued_operations.push_back(op);
+        info!(
+            "scheduled metadata upload for layer file uploads, ({} files uploaded)",
```

Comment:
```suggestion
            "scheduled metadata upload for layer file uploads, (after {} layer files to upload)",
```

They are most probably not really uploaded yet at this moment?

Reply:
Similar comment about `{} files removed` below.